From a149ad5ad855946203da62c6fc6a6f701d4907ae Mon Sep 17 00:00:00 2001
From: David Vernet <void@manifault.com>
Date: Fri, 1 Dec 2023 09:52:00 -0600
Subject: [PATCH 47/67] nest: Fix stats printing

We were only printing NR_STATS per-cpu, which is less than the actual
number of stats we have. We should be printing a total of NEST_STAT(NR)
stats, or the consumed stats won't be printed.

Signed-off-by: David Vernet <void@manifault.com>
---
 tools/sched_ext/scx_nest.bpf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/sched_ext/scx_nest.bpf.c b/tools/sched_ext/scx_nest.bpf.c
index 650a4af5a..315e88b04 100644
--- a/tools/sched_ext/scx_nest.bpf.c
+++ b/tools/sched_ext/scx_nest.bpf.c
@@ -102,7 +102,7 @@ struct {
 	__uint(type, BPF_MAP_TYPE_PERCPU_ARRAY);
 	__uint(key_size, sizeof(u32));
 	__uint(value_size, sizeof(u64));
-	__uint(max_entries, NR_STATS);
+	__uint(max_entries, NEST_STAT(NR));
 } stats SEC(".maps");
 
 
-- 
2.43.0.rc2

